src/types/model.ts

```diff
@@ -0,0 +1,187 @@
+import { DuosUserResponse } from './responseTypes';
+
+export type UserRoleName =
+  | 'Admin'
+  | 'Chairperson'
+  | 'Member'
+  | 'Researcher'
+  | 'Alumni'
+  | 'SigningOfficial'
+  | 'DataSubmitter'
+  | 'All';
+
+export interface UserRole {
+  roleId: number;
+  name: UserRoleName;
+  userId: number;
+  userRoleId: number;
+}
+
+export interface UserStatusInfo {
+  adminEnabled: boolean;
+  enabled: boolean;
+  userEmail: string;
+  userSubjectId: string;
+}
+
+export interface UserProperty {
+  eraAuthorized: string;
+  eraExpiration: string;
+  suggestedInstitution: string;
+  suggestedSigningOfficial: string;
+  selectedSigningOfficial: string;
+  daaAcceptance: string;
+}
+
+export interface DuosUser {
+  createDate: Date;
+  displayName: string;
+  email: string;
+  emailPreference: boolean;
+  isAdmin: boolean;
+  isAlumni: boolean;
+  isChairPerson: boolean;
+  isDataSubmitter: boolean;
+  isMember: boolean;
+  isResearcher: boolean;
+  isSigningOfficial: boolean;
+  libraryCards: LibraryCard[];
+  researcherProperties: UserProperty[];
+  roles: UserRole[];
+  userId: number;
+  userStatusInfo: UserStatusInfo;
+}
+
+export interface SimplifiedDuosUser {
+  userId: number;
+  displayName: string;
+  email: string;
+}
+
+export interface LibraryCard {
+  id: number;
+  userId: number;
+  institution: Institution;
+  institutionId: number;
+  eraCommonsId: string;
+  userName: string;
+  userEmail: string;
+  createDate: string;
+  createUserId: number;
+  updateDate: string;
+  updateUserId: number;
+}
+
+export type OrganizationType = 'For-Profit' | 'Nonprofit';
+
+export interface Institution {
+  id: number;
+  name: string;
+  itDirectorName: string;
+  itDirectorEmail: string;
+  institutionUrl: string;
+  dunsNumber: number;
+  orgChartUrl: string;
+  verificationUrl: string;
+  verificationFilename: string;
+  organizationType: OrganizationType;
+  createUser: number;
+  createDate: Date;
+  updateUser: number;
+  updateDate: Date;
+  signingOfficials: SimplifiedDuosUser[];
+}
+
+export interface DataSet {
+  name: string;
+  datasetName: string;
+  dataSetId: number;
+  createUserId: number;
+  createUser: DuosUserResponse;
+  dacId: string;
+  consentId: string;
+  translatedDataUse: string;
+  deletable: boolean;
+  properties: DatasetProperty[];
+  study: Study;
+  active: boolean;
+  needsApproval: boolean;
+  alias: string;
+  datasetIdentifier: string;
+  objectId: string;
+  /**
+   * TODO: Determine what type this is
+   */
+  dataUse: unknown;
```

Comment:
This is here: https://consent-ontology.dsde-dev.broadinstitute.org/#/Data%20Use/get_schemas_data_use_v4